lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/include/Config/Config.hpp

```diff
@@ -0,0 +1,181 @@
+#ifndef CONFIG_CLASS_HPP
+#define CONFIG_CLASS_HPP
+
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#pragma once
+
+#include <filesystem>
+#include <fstream>
+#include <sstream>
+#include <array>
+#include <string>
+
+#include "json.h"
+#include "include/logging.h"
+#include "include/io_helpers.h"
+#include "include/Config/Serializable.hpp"
+#include "include/Config/Version.hpp"
+
+namespace pew::eos::config
+{
+    /**
+     * \brief Used to describe information and functionality that is common to
+     * all Config classes.
+     */
+    struct Config : public Serializable
+    {
+        // Delete the copy constructor and copy assignment operator
+        Config(const Config&) = delete;
+        Config& operator=(const Config&) = delete;
+
+        /**
+         * \brief Gets the config class and values indicated by the template
+         * parameter given.
+         * \tparam T The Config-type-derived class that is being retrieved.
+         * \return An instance of a class derived from Config that contains all
+         * values.
+         */
+        template <typename T>
+        static std::enable_if_t<std::is_base_of_v<Config, T>, std::unique_ptr<T>> get()
+        {
+            // Create the config class
+            auto config = std::unique_ptr<T>(new T());
+
+            // Read the values from the file
+            config->read();
+
+            return config;
+        }
+
+    private:
+
+        // Depending on the configuration (debug or release) these are the possible relative paths to the config directory
+        static constexpr std::array<std::string_view, 2> s_possible_config_directories = {
```

Comment:
I agree completely. Part of the reason that the first entry in this array is the one that is utilized during release mode - so that in release the correct one is used automatically. A strong argument could be made for removing it altogether, since there are only two entries.